base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -133,21 +134,28 @@ impl<B: BlockchainBackend> Clone for Validators<B> {
 /// us to keep the reading and writing API extremely simple. Extending the types of data that the back ends can handle
 /// will entail adding to those enums, and the back ends, while this trait can remain unchanged.
 pub trait BlockchainBackend: Send + Sync {
-    /// Commit the transaction given to the backend. If there is an error, the transaction must be rolled back, and
-    /// the error condition returned. On success, every operation in the transaction will have been committed, and
+    /// Adds a block to the orphan database. On success, every operation in the transaction will have been committed,
+    /// and the function will return `Ok(())`.
+    fn add_orphan_block(&mut self, block: Block) -> Result<(), ChainStorageError>;
+    /// This function moves the block from the orphan pool to the main db. On success, every operation in the
+    /// transaction will have been committed, and the function will return `Ok(())`.
+    fn add_block(&mut self, block_hash: HashOutput) -> Result<(), ChainStorageError>;
+    // rewinds the database to the specified height. On success, every operation in the transaction will have been
+    // committed, and
     /// the function will return `Ok(())`.
-    fn write(&mut self, tx: DbTransaction) -> Result<(), ChainStorageError>;
+    fn rewind_to_height(&mut self, height: u64) -> Result<(), ChainStorageError>;
+    /// Commit the transaction given to the backend. This will create a db write transaction. This inly for syncing. On
+    /// success, every operation in the transaction will have been committed, and the function will return `Ok(())`.
+    fn write_sync(&mut self, tx: DbTransaction) -> Result<(), ChainStorageError>;
```

Comment:
I think since we're going down the road of individual API functions in this trait with explicit contracts, we can think about doing away with write_sync and DbTransaction completely

Reply:
I was going down this route as a sync call, as when you download/ sync data, you only ever add data directly to the db as you would have downloaded them. This is more inline with the new pruning sync. You would download the data in packs, like first headers, then kernels etc. There is no more blocks. So rather make a sync function to pipe data directly into the db with almost no extra logic. Thats my thinking behind this function.